pandas/tests/indexing/test_loc.py

```diff
@@ -712,6 +712,27 @@ def test_loc_assign_non_ns_datetime(self, unit):
         expected = Series(df.loc[:, "expected"], name=unit)
         tm.assert_series_equal(df.loc[:, unit], expected)
 
+    def test_loc_modify_datetime(self):
+        # GH 28837
+        df = pd.DataFrame.from_dict(
+            {"date": [1485264372711, 1485265925110, 1540215845888, 1540282121025]}
+        )
+        df["date_dt"] = pd.to_datetime(df["date"], unit="ms", cache=True)
+        df.loc[:, "date_dt_cp"] = df.loc[:, "date_dt"]
+        df.loc[[2, 3], "date_dt_cp"] = df.loc[[2, 3], "date_dt"]
+        result = df
+
+        data = """date,date_dt,date_dt_cp
```

Comment:
rather than using read_csv, you can put the dates into an array above (line 718) and then just construct it programatically.

Reply:
The text stored in variable `data` is in CSV format. How do I store it in an array? 

I referred to the following code from the same file: 
https://github.com/pandas-dev/pandas/blob/a6f9593558441ed98e6886d129e4ee1f7abe5e5f/pandas/tests/indexing/test_loc.py#L665-L673

Reply:
just construct the DataFrame directly, you *already* have a df with data in it.

Reply:
@rohitsanj : One reason for using `read_csv` in that example is that it's easier to construct the `MultiIndex` that makes up the column and row headers.  In your case, your setup is a lot simpler, hence why it's preferred that you then construct the `DataFrame` explicitly.

Reply:
Taking this one home.  Comment addressed.